src/components/Text/TextStyles.ts

```diff
@@ -0,0 +1,38 @@
+import { TextStyle } from 'react-native';
+import { Theme } from '../BrandConfigProvider/Theme';
+
+export type Variant = 'body' | 'heading' | 'subHeading';
+
+export class TextStyles {
+  body: TextStyle = {};
+  heading: TextStyle = {};
+  subHeading: TextStyle = {};
+
+  private _variants: Variant[] = ['body', 'heading', 'subHeading'];
+
+  constructor(theme: Theme) {
+    const defaultStyles: Partial<TextStyles> = {
+      body: {
+        color: theme.colors.text,
+      },
+      heading: {
+        color: theme.colors.text,
+        fontSize: 24,
+        fontWeight: 'bold',
+      },
+      subHeading: {
+        color: theme.colors.text,
+        fontSize: 18,
+        fontWeight: 'bold',
+      },
+    };
+
+    this.mergeStyles(defaultStyles);
+  }
+
+  mergeStyles(styles: Partial<TextStyles>) {
+    this._variants.forEach((variant) => {
+      Object.assign(this[variant], styles[variant]);
+    });
```

Comment:
I'm trying to decide how I feel about each component needing its own `*Styles` class.  I especially don't know if I like each having to implement a `mergeStyles` like this.  I like how only the component can/should know how it can merge styles into its internal styles representation - it's the class & method I'm not sure about.  For example, if I extrapolate this out into 10s of components, a consumer might have to do something like this in order to brand the app:
```
... create theme
const styles = new Styles({ theme });
styles.Text.mergeStyles(customTextStyles);
styles.Component2.mergeStyles(component2StyleOverrides);
styles.Component3.mergeStyles(component3StyleOverrides);
styles.Component4.mergeStyles(component4StyleOverrides);
styles.Component5.mergeStyles(component5StyleOverrides);
styles.Component6.mergeStyles(component6StyleOverrides);
... 10-15 more

return (<BrandConfigProvider styles={styles}>...)
```

I wonder if we should get ahead of that and make `mergeStyles` more of an impl detail that consumers don't need to worry about?  Something as simple as moving that merge logic to the ctor might go a long way:
```
... create theme
const styleOverrides = {
  Text: { body: { color: 'red' } },
  Component2: ...
  Component3: ...
};
const styles = new Styles({ theme, styleOverrides });
return (<BrandConfigProvider styles={styles}>...)
```
Not necessarily less LOC, just less method calls in my mind.

Reply:
I think we are thinking along the same lines here.  I'm sure we have not arrived on the final API from a customer perspective and I don't expect `.megeStyles()` to be part of that.  Just building with iteration and creating simple primitives.

My next PR will have more on this, but in short I'm thinking that we'll want to support a number of use cases from just tweaking a few colors, to redesigning every component to match a consumer brand.